File: semester4/web-programming/asp-dotnet/Models/Recipe.cs
namespace asp_dotnet.Models
{
    public class Recipe
    {
        public int RecipeID { get; set; }
        public string Name { get; set; }
        public string Author { get; set; }
        public string Type { get; set; }
        public string Instructions { get; set; }
        public DateOnly Date { get; set; }
    }
}

File: semester4/web-programming/assignments/angular-php/Server/Model/Recipe.php
<?php
    class Recipe {
        public $RecipeID;
        public $Name; 
        public $Author;
        public $Type;
        public $Instructions;
        public $Date;

        public function isValid() {
            return !empty($this->Name) && 
                !empty($this->Author) && 
                !empty($this->Type) && 
                !empty($this->Instructions) &&
                !empty($this->Date);
        }
    }
?>